rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -234,13 +246,53 @@ error_code cellSearchStartListSearch(CellSearchListSearchType type, CellSearchSo
 	{
 		auto curr_search = idm::get<search_object_t>(id);
 		vm::var<CellSearchResultParam> resultParam;
-		resultParam->searchId  = id;
+		resultParam->searchId = id;
 		resultParam->resultNum = 0; // Set again later
 
 		std::function<void(const std::string&)> searchInFolder = [&, type](const std::string& vpath)
 		{
 			// TODO: this is just a workaround. On a real PS3 the playlists seem to be stored in dev_hdd0/mms/db/metadata_db_hdd
-			for (auto&& item : fs::dir(vfs::get(vpath)))
+
+			std::vector<fs::dir_entry> dirs_sorted;
+
+			for (auto&& entry : fs::dir(vfs::get(vpath)))
+			{
+				entry.name = vfs::unescape(entry.name);
+
+				if (entry.is_directory)
+				{
+					if (entry.name == "." || entry.name == "..")
+					{
+						continue; // these dirs are not included in the dir list
+					}
+
+					dirs_sorted.push_back(entry);
+				}
+			}
+
+			// clang-format off
+			std::sort(dirs_sorted.begin(), dirs_sorted.end(), [&](const fs::dir_entry& a, const fs::dir_entry& b) -> bool
```

Comment:
Since you seem to be handling playlists by folder and not by using a yaml file, this approach should be fine for now